includes/class-wc-stripe-blocks-support.php

```diff
@@ -81,26 +93,55 @@ public function get_payment_method_script_handles() {
 	 * @return array
 	 */
 	public function get_payment_method_data() {
-		return [
-			'stripeTotalLabel'    => $this->get_total_label(),
-			'publicKey'           => $this->get_publishable_key(),
-			'allowPrepaidCard'    => $this->get_allow_prepaid_card(),
-			'title'               => $this->get_title(),
-			'button'              => [
-				'type'        => $this->get_button_type(),
-				'theme'       => $this->get_button_theme(),
-				'height'      => $this->get_button_height(),
-				'locale'      => $this->get_button_locale(),
-				'customLabel' => isset( $this->settings['payment_request_button_label'] ) ? $this->settings['payment_request_button_label'] : 'Buy now',
-			],
-			'inline_cc_form'      => $this->get_inline_cc_form(),
-			'icons'               => $this->get_icons(),
-			'showSavedCards'      => $this->get_show_saved_cards(),
-			'showSaveOption'      => $this->get_show_save_option(),
-			'supports'            => $this->get_supported_features(),
-			'stripeLocale'        => WC_Stripe_Helper::convert_wc_locale_to_stripe_locale( get_locale() ),
-			'isAdmin'             => is_admin(),
-		];
+		// We need to call array_merge_recursive so the blocks 'button' setting doesn't overwrite
+		// what's provided from the gateway or payment request configuration.
+		return array_merge_recursive(
+			$this->get_gateway_javascript_params(),
+			$this->get_payment_request_javascript_params(),
+			// Blocks-specific options
+			[
+				'title'          => $this->get_title(),
+				'icons'          => $this->get_icons(),
+				'supports'       => $this->get_supported_features(),
+				'showSavedCards' => $this->get_show_saved_cards(),
+				'showSaveOption' => $this->get_show_save_option(),
+				'isAdmin'        => is_admin(),
+				'button'         => [
+					'customLabel' => $this->payment_request_configuration->get_button_label(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Returns the Stripe Payment Gateway JavaScript configuration object.
+	 *
+	 * @return array  the JS configuration from the Stripe Payment Gateway.
+	 */
+	private function get_gateway_javascript_params() {
+		$js_configuration = [];
+
+		$gateways = WC()->payment_gateways->get_available_payment_gateways();
+		if ( isset( $gateways['stripe'] ) ) {
+			$js_configuration = $gateways['stripe']->javascript_params();
```

Comment:
Hm, looks like the `get_supported_features` method below just instantiates a `new WC_Gateway_Stripe()`… think we should change one or the other for consistency, or leave them as they are?

Reply:
Nice catch!

Fixed in 0c9e774c82853a3cfcac3844031f59609ad981a1